android/src/main/java/com/mattermost/networkclient/APIClientModule.kt

```diff
@@ -4,99 +4,124 @@ import com.facebook.react.bridge.*
 import okhttp3.OkHttpClient
 import okhttp3.Request
 import okhttp3.RequestBody.Companion.toRequestBody
-import com.facebook.react.bridge.Arguments;
+import okio.IOException
 
 class APIClientModule(reactContext: ReactApplicationContext) : ReactContextBaseJavaModule(reactContext) {
 
-    var client = OkHttpClient();
+    var sessionsClient = mutableMapOf<String, OkHttpClient.Builder>();
     var sessions = mutableMapOf<String, Request.Builder>()
 
     override fun getName(): String {
         return "APIClient"
     }
 
     @ReactMethod
-    fun createClientFor(baseUrl: String, config: ReadableMap, promise: Promise){
+    fun createClientFor(baseUrl: String, options: ReadableMap, promise: Promise) {
         try {
+            // Create the client and request builder
+            sessionsClient[baseUrl] = OkHttpClient().newBuilder();
             sessions[baseUrl] = Request.Builder().url(baseUrl);
-            promise.resolve(sessions[baseUrl]);
-        } catch (err: Throwable){
+
+            // Attach client options
+            sessionsClient[baseUrl]?.parseOptions(options);
+
+            // Return stringified client for success
+            promise.resolve(sessions[baseUrl].toString())
```

Comment:
Not following what sort of string is sent back when the promise gets resolved, can you clarify

Reply:
Just the name of the class - but I realise I can pass back kotlin's `Unit` which is essentially `void` in java (`promise.resolve(val)` always expected a `val` so I had to set something...).